src/Common/BufferAllocationPolicy.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "config.h"
+
+#include "logger_useful.h"
+
+#include <list>
+
+namespace DB
+{
+
+struct BufferAllocationSettings
+{
+    size_t strict_upload_part_size = 0;
+    size_t min_upload_part_size = 16 * 1024 * 1024;
+    size_t max_upload_part_size = 5ULL * 1024 * 1024 * 1024;
+    size_t upload_part_size_multiply_factor = 2;
+    size_t upload_part_size_multiply_parts_count_threshold = 500;
+    size_t max_single_part_upload_size = 32 * 1024 * 1024;
+};
+
+class IBufferAllocationPolicy
```

Comment:
I feel it's better to say in the name that this allocation policy is for growing buffers:
`IBufferAllocationPolicy` -> `GrowingBufferAllocationPolicy`

Reply:
I think IBufferAllocationPolicy is better, as there are 2 types, Fixed and Exp. Only Exp means Growing BufferAllocationPolicy the other one is Fixed.

Reply:
My idea was to say in the name of the class that it's an allocation policy for series of buffers, not just for one specific buffer.

Reply:
ya I understand, `GrowingOrFixedBufferAllocationPolicy` is too long, but removed Fixed makes it a specification. 
How about BufferSeriesAllocationPolicy ?

Reply:
I agree, `BufferSeriesAllocationPolicy` looks better.